MapboxNavigation/NavigationMapView.swift

```diff
@@ -104,27 +104,24 @@ open class NavigationMapView: MGLMapView, UIGestureRecognizerDelegate {
     
     open override var showsUserLocation: Bool {
         get {
-            if tracksUserCourse || userLocationForCourseTracking != nil {
+            if tracksUserCourse {
                 return !(userCourseView?.isHidden ?? true)
             }
             return super.showsUserLocation
         }
         set {
-            if tracksUserCourse || userLocationForCourseTracking != nil {
-                super.showsUserLocation = false
-                
+            super.showsUserLocation  = !tracksUserCourse
```

Comment:
With this change, `mapView.showsUserLocation = true` and `mapView.showsUserLocation = false` have the same effect on `MGLMapView.showsUserLocation` – `newValue` is ignored.

Instead of overriding the setter, define a `didSet` on `showsUserLocation`:

```swift
didSet {
    if showsUserLocation {
        // showsUserLocation and tracksUserCourse cannot both be true at the same time.
        tracksUserCourse = false
    }
    if showsUserLocation && userTrackingMode == .course {
        if userCourseView == nil { … }
        userCourseView?.isHidden = false
    } else {
        userCourseView?.isHidden = true
    }
}
```

then rename `tracksUserCourse` to `showsUserCourse` to avoid confusion with `userTrackingMode`.